src/samples/design_patterns/Makefile

```diff
@@ -0,0 +1,2 @@
+sample-$(DESIGN_PATTERN_SAMPLES) += stream_sample
+sample-stream_sample-$(DESIGN_PATTERN_SAMPLES) := stream_sample.c
```

Comment:
To be honest I prefer as is. We can add more design pattern samples in the future